Neos.ContentRepository.Core/Classes/Service/ContentStreamPruner.php

```diff
@@ -76,30 +105,37 @@ public function prune(bool $removeTemporary = false): iterable
      *
      *   - Otherwise, we cannot replay the other content streams correctly (if the base content streams are missing).
      *
-     * @return ContentStreams the removed content streams
+     * @return list<ContentStreamId> the removed content streams
      */
-    public function pruneRemovedFromEventStream(): ContentStreams
+    public function pruneRemovedFromEventStream(): array
```

Comment:
todo maybe one wants full control over what will be removed and why? Should we introduce a dry mode? So one calls it once dry, then check and than reruns ... or we could backup the events table?

Reply:
I'd  say dry run seems fine?

Reply:
build a little `contentstream:status` utility, and will add a warning to the prune commands that you might need to check with `contentstream:status` first and they should not be run if the system is in use as then a "forked" content stream might really appear for just a second ...

```
flow contentstream:status                     
Content streams that are not removed (ContentStreamWasRemoved) and not in use by workspace:
  id: 81a50ad4-41bb-43b9-8e2a-fac6b9beb3ee reason for removal: forked (temporary state)
  id: 0db932cb-13bc-4f38-a4de-d0c304c7305f reason for removal: forked (temporary state)
  id: 6ddb063c-ff46-4a12-8bc2-7bd701567b3b reason for removal: forked (temporary state)
  id: 3058560a-348e-497b-93a5-95418d49a54a reason for removal: forked (temporary state)
  id: 2a36581d-3c40-4289-99a7-deba45a052ef reason for removal: no longer in use
To remove the dangeling streams from the projections please run ./flow contentstream:prune
Then they are ready for removal from the event store

Removed content streams that can be pruned from the event store
  id: 310d09ef-4b5e-412c-9c08-9602ef05b85e previous state: no longer in use
  id: 5aa33bdd-95c2-4db3-98c6-5fd270a128b1 previous state: forked (temporary state)
To prune the removed streams from the event store run ./flow contentstream:pruneremovedfromeventstream
Then they are indefinitely pruned from the event store
```